blog/_posts/2023-05-20-sbt-community-repository-follow-up.md

```diff
@@ -0,0 +1,38 @@
+---
+layout: blog-detail
+post-type: blog
+by: Julien Richard-Foy, Scala Center
+title: "sbt Community Repository (Follow-Up)"
+---
+
+[Last month]({% link blog/_posts/2023-04-20-sbt-plugins-community-repository.md %}), I published a blog article where I described the sbt community repository. In this follow-up article, I present our plan to strengthen the infrastructure that supports the sbt ecosystem.
+
+This blog article is also an initial response to the related Advisory Board proposal [SCP-029 sbt community repository](https://github.com/scalacenter/advisoryboard/blob/main/proposals/029-sbt-community-repository.md), which has been approved by the Advisory Board during the meeting of the [27th of April](https://github.com/scala/scala.epfl.ch/blob/main/minutes/_posts/2023-04-27-april-27-2023.md).
+
+## Current State
+
+The sbt community repository was set up a couple of years ago with two primary purposes:
+1. host “old” versions of sbt plugins so that “old” Scala projects continue to work today and in the future,
+2. give some time to the ecosystem of sbt plugins to migrate to Maven Central.
+
+On average, the community repository serves 6 TB of data per month and 700 requests per second. These numbers are not huge, but I think they are higher than they should be, especially when we take into account the fact that the content delivered by the community repository is usually downloaded once per client and then cached forever. I would like to thank again JFrog for their continued support of the sbt project and the Scala ecosystem.
+
+The reason why these numbers are so high is that the migration to Maven Central did not completely happen and many sbt plugins still have their latest stable version on the community repository.
+As a consequence, any new developer joining a Scala project or any continuous integration system building a Scala project with a fresh cache will probably download artifacts from the community repository, even if the project has up-to-date dependencies.
+
+The event of the [7th of April]({% link blog/_posts/2023-04-20-sbt-plugins-community-repository.md %}) has shown that the sbt community repository plays a critical role in the ecosystem. Therefore, we need to increase our safety margins.
+
+## Action Plan
+
+Our goal is to ensure all the Scala projects can be built in the future regardless of which versions of sbt plugins they use.
+
+To achieve this, we will:
+
+1. Accelerate the migration of the actively used sbt plugins to Maven Central. This will reduce the load on the community repository and make it easier to host mirrors of it. We will need the help of the plugin authors to make progress on this item.
```

Comment:
```suggestion
1. Accelerate the migration of the actively used sbt plugins to Maven Central. This will reduce the load on the community repository and make it easier to host mirrors of it. We will need the help of the plugin authors to make this happen, and also help from the community in identifying these plugins.
```

Reply:
I skipped the second part as for now we will see if we can get somewhere already by analyzing the HTTP server logs.